src/test/java/test/methodselection/MethodSelectionTest.java

```diff
@@ -0,0 +1,65 @@
+package test.methodselection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.testng.annotations.Test;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+import test.InvokedMethodNameListener;
+import test.SimpleBaseTest;
+
+public class MethodSelectionTest extends SimpleBaseTest {
```

Comment:
Should we also add a test which includes a `$` and a wild card character and ensuring that all methods that have `$` get executed ?